gcc/rust/rust-session-manager.cc

```diff
@@ -474,6 +526,28 @@ Session::enable_dump (std::string arg)
 void
 Session::parse_files (int num_files, const char **files)
 {
+  rust_assert (num_files > 0);
+
+  if (options.crate_name.empty ())
+    {
+      /* HACK: We use the first file to infer the crate name, which might be
+       * incorrect: since rustc only allows one file to be supplied in the
+       * command-line */
+      auto crate_name = infer_crate_name (files[0]);
+      rust_debug_loc (Location (), "inferred crate name: %s",
+		      crate_name.c_str ());
+      if (!options.set_crate_name (crate_name))
+	{
+	  // fake a linemapping so that we can show the filename
+	  linemap->start_file (files[0], 0);
+	  linemap->start_line (0, 1);
+	  rust_inform (linemap->get_location (0),
+		       "crate name inferred from this file");
+	  linemap->stop ();
```

Comment:
we don't need this linemap, since the options.set_crate_name() calls the validate function errors are already emited.

Reply:
> we don't need this linemap, since the options.set_crate_name() calls the validate function errors are already emited.

I added this due to UX consideration, so the user will know which file caused this issue (the crate name is derived from the filename after all).